java/com/google/flatbuffers/ArrayReadWriteBuf.java

```diff
@@ -240,6 +240,13 @@ public boolean requestCapacity(int capacity) {
     // implemented in the same growing fashion as ArrayList
     int oldCapacity = buffer.length;
     int newCapacity = oldCapacity + (oldCapacity >> 1);
+    if (newCapacity - capacity <= 0) {
```

Comment:
why does this have a separate case for `0`? surely that is also an error?

Reply:
Sorry for not explaining better. This is written under the assumption that `capacity` might have "looped around"/overflown and became negative because it exceeded `Integer.MAX_VALUE`. E.g. the method is called like `requestCapacity(index + 8)`: if `index` is `Integer.MAX_VALUE` the `+8` will result in `capacity` being `Integer.MIN_VALUE + 8`.

The condition will catch two cases:
- `newCapacity` is not large enough,
- the requested `capacity` has overflown/is negative.

The second condition (`capacity < 0`) intercepts the second case (overflow of requested capacity) and throws.

Reply:
The code looks correct, although the conditions look a bit strange. Maybe comment to make it more clear what is happening.

Reply:
LGTM